### DISCUSSION TITLE
Bad request 400

### DISCUSSION THREAD

**Discussion by ffabros** [netbox-community/netbox#9777] [General]
Hi folks,

I've been in trouble with HTTPS.

I can connect using develop mode port 8000. But when I am trying to connect using 8001 it returns an error.
--------------------
Gunicorn config:

  GNU nano 6.2                                                                   /opt/netbox/gunicorn.py                                                                            
# The IP address (typically localhost) and port that the Netbox WSGI process should listen on
bind = '127.0.0.1:8001'

# Number of gunicorn workers to spawn. This should typically be 2n+1, where
# n is the number of CPU cores present.
workers = 5

# Number of threads per worker process
threads = 3

# Timeout (in seconds) for a request to complete
timeout = 120

# The maximum number of requests a worker can handle before being respawned
max_requests = 5000
max_requests_jitter = 500
----------------------------------
etc/Nginx/sites-enabled config:

 # CHANGE THIS TO YOUR SERVER'S NAME
    server_name _;

    ssl_certificate /etc/ssl/certs/netbox.crt;
    ssl_certificate_key /etc/ssl/private/netbox.key;

    client_max_body_size 25m;

    location /static/ {
        alias /opt/netbox/netbox/static/;
    }

    location / {
        proxy_pass http://127.0.0.1:8001;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}

server {
    # Redirect HTTP traffic to HTTPS
   listen [::]:80 ipv6only=off;
   server_name _;
  return 301 https://$host$request_uri;
}

I truly appreciate any help on this.

Thank you.

**Comment by candlerb** [comment]
Firstly, please edit your post and add three backticks (` ``` `) on a line of their own before and after the posted configs.  This will make them format properly.

> when I am trying to connect using 8001 it returns an error.

That is correct.  You have configured gunicorn to listen on localhost (127.0.0.1) port 8001.  The only client which can connect to it on that port is another process running on the same box - you cannot connect to it over the network on port 8001.

What you should be connecting to is your nginx server, which will be on `https://some.domain.name` which resolves to the IP address of your Netbox server (or you can try the IP address, but you'll definitely get a certificate error)
